articles/150_ros_command_line.md

```diff
@@ -0,0 +1,107 @@
+---
+layout: default
+title: ROS Command Line
+permalink: articles/ros_command_line.html
+abstract:
+  This article describes ROS 2 nodes command line capabilities and its syntax.
+author: '[Michel Hidalgo](https://github.com/hidmic)'
+published: true
+---
+
+- This will become a table of contents (this text will be scraped).
+{:toc}
+
+# {{ page.title }}
+
+<div class="abstract" markdown="1">
+{{ page.abstract }}
+</div>
+
+Original Author: {{ page.author }}
+
+## Overview
+
+As it was the case in ROS 1, ROS 2 nodes allow configuration via command line arguments to a certain degree.
+However, a larger set of configuration options, an ambiguity in remapping rules and parameter assignment syntax (as a result of the leading underscore name convention for hidden resources), a one-to-many relationship between executables and nodes, among others, add up complexity to an otherwise simple interface.
+
+Increasingly precise addressing mechanisms, as well as leading double underscores (`__`) in positional arguments, solve part of the problem.
+In ROS 2, however, command line options are introduced too as these are:
+
+- Explicit and less error prone.
+- Suitable for tab completion.
+
+Unfortunately, additional devices must be put in place as these coexist with user-defined options -- part of the rationale behind ROS 1 command line design.
+
+## Namespacing
+
+To prevent ROS specific command line options from interacting with user-defined ones, the former are scoped using the `--ros-args` option and a trailing double dash token (`--`):
+
+```sh
+ros2 run some_package my_node [<user-defined-pre-arg0>...] \
+  --ros-args <ros-specific-arg-0> [<ros-specific-arg-1>...] -- <user-defined-arg-1> [<user-defined-arg-2>...]
+```
+
+If no user defined arguments are provided after ROS specific arguments are, the double dash token (`--`) may be elided:
+
+```sh
+ros2 run some_package my_node [<user-defined-arg-0>...] --ros-args <ros-specific-arg-0> [<ros-specific-arg-1>...]
+```
```

Comment:
One more thing to clarify, can you have more than one instance of `--ros-args ... --`? I would say we should so that multiple sources of ROS options can just append them to the command line arguments without having to worry if there was a previous instance or not.

Reply:
That's a very good point. See beaf7bec3dfc45db907f8c89e30801dcfdc3eba1.